source/common/stats/histogram_impl.cc

```diff
@@ -79,5 +79,64 @@ void HistogramStatisticsImpl::refresh(const histogram_t* new_histogram_ptr) {
   }
 }
 
+ParentHistogramImpl::ParentHistogramImpl(const std::string& name, Store& parent,
+                                         std::string&& tag_extracted_name, std::vector<Tag>&& tags)
+    : MetricImpl(std::move(tag_extracted_name), std::move(tags)), parent_(parent),
+      interval_histogram_(hist_alloc()), cumulative_histogram_(hist_alloc()),
+      interval_statistics_(interval_histogram_), cumulative_statistics_(cumulative_histogram_),
+      used_(false), name_(name) {}
+
+ParentHistogramImpl::~ParentHistogramImpl() {
+  hist_free(interval_histogram_);
+  hist_free(cumulative_histogram_);
+}
+
+void ParentHistogramImpl::recordValue(uint64_t value) {
+  used_ = true;
+  hist_insert_intscale(interval_histogram_, value, 0, 1);
+  parent_.deliverHistogramToSinks(*this, value);
+}
+
+void ParentHistogramImpl::merge() {
+  hist_accumulate(cumulative_histogram_, &interval_histogram_, 1);
+  interval_statistics_.refresh(interval_histogram_);
+  cumulative_statistics_.refresh(cumulative_histogram_);
+  hist_clear(interval_histogram_);
+}
+
+// TODO(mergeconflict): figure out the best way to eliminate duplication between this and
+// bucketSummary below, and the implementations in ThreadLocalParentHistogramImpl.
+const std::string ParentHistogramImpl::quantileSummary() const {
+  if (used()) {
+    std::vector<std::string> summary;
+    const std::vector<double>& supported_quantiles_ref = interval_statistics_.supportedQuantiles();
+    summary.reserve(supported_quantiles_ref.size());
+    for (size_t i = 0; i < supported_quantiles_ref.size(); ++i) {
+      summary.push_back(fmt::format("P{}({},{})", 100 * supported_quantiles_ref[i],
+                                    interval_statistics_.computedQuantiles()[i],
+                                    cumulative_statistics_.computedQuantiles()[i]));
+    }
+    return absl::StrJoin(summary, " ");
+  } else {
+    return std::string("No recorded values");
```

Comment:
nit: you can return "No recorded values" here rather than explicitly constructing the string.

Reply:
This is what I get for copy-pasting code :)